### DISCUSSION TITLE
JVM Metrics

### DISCUSSION THREAD

**Discussion by rupinder10** [open-telemetry/opentelemetry-java#2966] [Q&A]
Does anybody have a sample to convert java heap memory stats to metrics ? I have seen an example where there is a callback to MemoryMXBean to get the MemoryUsage objects. In my case, I cannot use that as I already have a MemoryUsage object passed to my code and I want to convert to metrics and send those metrics.

**Comment by jkwatson** [comment]
Something like this?  

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/MemoryPools.java

I know it uses the MemoryMXBean, but it's a pretty good example of generating this kind of metrics.
